src/fu-quirks.h

```diff
@@ -22,18 +23,112 @@
 #ifndef __FU_QUIRKS_H
 #define __FU_QUIRKS_H
 
-typedef struct {
-	const gchar             *sys_vendor;
-	const gchar		*identifier;
-	AsVersionParseFlag       flags;
-} FuVendorQuirks;
+G_BEGIN_DECLS
 
-static const FuVendorQuirks quirk_table[] = {
-	/* Dell & Alienware use AA.BB.CC.DD rather than AA.BB.CCDD */
-	{ "Dell Inc.",	"com.dell.uefi",	AS_VERSION_PARSE_FLAG_NONE },
-	{ "Alienware",	"com.dell.uefi",	AS_VERSION_PARSE_FLAG_NONE },
-	{ NULL,		NULL,			AS_VERSION_PARSE_FLAG_NONE }
-};
+#include <glib-object.h>
+#include <gusb.h>
 
-#endif /* __FU_QUIRKS_H */
+#define FU_TYPE_QUIRKS (fu_quirks_get_type ())
+G_DECLARE_FINAL_TYPE (FuQuirks, fu_quirks, FU, QUIRKS, GObject)
+
+FuQuirks	*fu_quirks_new				(void);
+gboolean	 fu_quirks_load				(FuQuirks	*self,
+							 GError		**error);
+const gchar	*fu_quirks_lookup_by_id			(FuQuirks	*self,
+							 const gchar	*prefix,
+							 const gchar	*id);
+const gchar	*fu_quirks_lookup_by_glob		(FuQuirks	*self,
+							 const gchar	*prefix,
+							 const gchar	*glob);
+const gchar	*fu_quirks_lookup_by_usb_device		(FuQuirks	*self,
+							 const gchar	*prefix,
+							 GUsbDevice	*dev);
+
+/**
+ * FU_QUIRKS_PLUGIN_DFU:
+ * @key: the USB device ID, e.g. `USB\VID_0763&PID_2806`
+ * @value: a string, separated using `|`, e.g. `ignore-polltimeout|no-pid-change`
+ *
+ * Assigns optional quirks to use for a DFU device which does not follow the
+ * DFU 1.0 or 1.1 specification.
+ *
+ * Default value: `none`
```

Comment:
So I think this should point to `DfuDeviceQuirks` in plugins/dfu/dfu-device.h, but those strings aren't really discoverable right now.

Reply:
I think they probably need documenting (fully) in this gtk-doc block. Maybe in a list?